File: leetcode/01-two-sum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

class Solution {
 public:
  std::vector<int> twoSum(std::vector<int> &nums, int target) {
    std::unordered_map<int, int> valToIndex = {};

    for (int i = 0; i < nums.size(); ++i) {
      if (auto j = valToIndex.find(target - nums[i]); j != valToIndex.end()) {
        return {j->second, i};
      }
      valToIndex.insert({nums[i], i});
    }
    return {};
  }
};

int main() {}

File: leetcode/go/01-two-sum.go
package main

func twoSum(nums []int, target int) []int {
	var valToIndex = make(map[int]int)

	for i := 0; i < len(nums); i++ {
		if value, mapContains := valToIndex[target-nums[i]]; mapContains {
			return []int{value, i}
		}
		valToIndex[nums[i]] = i
	}

	return []int{}
}
